ui/components/app/assets/nfts/nft-grid/nft-grid.tsx

```diff
@@ -9,8 +9,56 @@ import { NftItem } from '../../../../multichain/nft-item';
 import { NFT } from '../../../../multichain/asset-picker-amount/asset-picker-modal/types';
 import {
   getCurrentNetwork,
+  getIpfsGateway,
   getNftIsStillFetchingIndication,
 } from '../../../../../selectors';
+import useGetAssetImageUrl from '../../../../../hooks/useGetAssetImageUrl';
+
+const NFTGridItem = (props: {
+  nft: NFT;
+  onClick: () => void;
+  privacyMode?: boolean;
+  currentChain: {
+    chainId: Hex;
+    nickname: string;
+    rpcPrefs?: {
+      imageUrl: string;
+    };
+  };
+}) => {
+  const { nft, onClick, privacyMode, currentChain } = props;
+
+  const { image, imageOriginal } = nft;
+
+  const ipfsGateway = useSelector(getIpfsGateway);
+  const nftImageURL = useGetAssetImageUrl(
+    imageOriginal ?? image ?? undefined,
+    ipfsGateway,
+  );
+
+  const isImageHosted =
+    image?.startsWith('https:') || image?.startsWith('http:');
+  const nftItemSrc = isImageHosted ? image : nftImageURL;
+
+  const nftImageAlt = getNftImageAlt(nft);
+
+  const nftSrcUrl = imageOriginal ?? image;
+  const isIpfsURL = nftSrcUrl?.startsWith('ipfs:');
```

Comment:
This logic now matches the NFT Details page.

NOTE - we should probably refactor and reuse this logic. But for cherry-picks, we can keep it here.

Reply:
The main fix was this `isIpfsURL` condition, as it seemed that the GridView before treated the IPFS url as a normal URL and attempted to fetch and display it.

Now the logic matches the details page, this has been resolved.